app/controllers/volunteers_controller.rb

```diff
@@ -26,7 +27,15 @@ def create
 
   def update
     if @volunteer.update(volunteer_params)
-      redirect_to @volunteer, notice: t('volunteer_updated')
+      if @state == 'interested/registered' && @volunteer.state == 'active'
+        new_user = User.new(email: @volunteer.email,
+          password: Devise.friendly_token, role: 'volunteer')
+        new_user.save!
+        new_user.invite!
```

Comment:
Why is the password stuff needed? In the users_controller, they just get invited without setting password.
Also i don't see here that the volunteer gets associated to the user_account, something like:
```ruby
@volunteer.build_user(email: @volunteer.email)
```
And all this should be done before `if @volunteer.update(volunteer_params)`.